### PR TITLE
refactor(sheet): copy&paste

### PR DESCRIPTION
**Pull request by waterhydroxyl**
1. Remove the use of getComputedStyle API to improve the paste parsing speed
2. Keep the style data of each cell in the merged cell consistent when pasting, and improve the consistency of the direct copy and paste scene with Excel and Google Sheets

## Pull Request Checklist

- [x] Related tickets or issues have been linked in the PR description (or missing issue).
- [x] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [x] Unit tests have been added for the changes (if applicable).
- [x] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Comment by waterhydroxyl** [comment]
The use of DOMParser was removed based on this [example](https://stackoverflow.com/questions/56451731/dom-parser-chrome-extension-memory-leak).

File: packages/sheets-ui/src/services/clipboard/html-to-usm/converter.ts [hunk 1]
```diff
@@ -230,18 +224,63 @@ export class HtmlToUSMService {
...
+                    this._getStyleBySelectorText(node.nodeName.toLowerCase(), key) || this._getStyleBySelectorText(node.nodeName, 'text-decoration-line') || recordStyle['text-decoration-line'] || '';
+                textDecoration && (newStyleStr += `text-decoration:${textDecoration};`);
+                continue;
+            }
+
+            const value =
+                style.getPropertyValue(key) ||
+                this._getStyleBySelectorText(`.${node.className}`, key) ||
+                this._getStyleBySelectorText(node.nodeName.toLowerCase(), key) ||
+                recordStyle[key] ||
```
Comment (ybzky) [inline packages/sheets-ui/src/services/clipboard/html-to-usm/converter.ts:272]: here need to consider classList and id，

File: packages/sheets-ui/src/services/clipboard/html-to-usm/converter.ts [hunk 2]
```diff
@@ -101,40 +92,43 @@ export class HtmlToUSMService {
...
+        const style = this._dom.querySelector('style');
+        if (style) {
+            document.head.appendChild(style);
+            for (const rule of style.sheet!.cssRules) {
+                const cssRule = rule as CSSStyleRule;
+                const selectorText = cssRule.selectorText;
+                const style = cssRule.style;
+                this._styleMap.set(selectorText, style);
+            }
+            style.remove();
```
[the remark this answers is not in the document]
Comment (waterhydroxyl) [reply packages/sheets-ui/src/services/clipboard/html-to-usm/converter.ts:130]: That's indeed a problem. I am inclined to use Shadow DOM to solve this issue.
```typescript
if (style) {
    const shadowHost = document.createElement('div');
    const shadowRoot = shadowHost.attachShadow({ mode: 'open' });
    document.body.appendChild(shadowHost);
    shadowRoot.appendChild(style);

    for (const rule of style.sheet!.cssRules) {
        const cssRule = rule as CSSStyleRule;
        const selectorText = cssRule.selectorText;
        const style = cssRule.style;
        this._styleMap.set(selectorText, style);
    }
    style.remove();
    shadowHost.remove();
}
```

File: packages/sheets-ui/src/services/clipboard/html-to-usm/converter.ts (hunk 1 above)
Comment (waterhydroxyl) [reply packages/sheets-ui/src/services/clipboard/html-to-usm/converter.ts:272]: I used Excel, Tencent Docs, and Google Sheets for pasting operations, and I haven't found any cell nodes using multiple class names or ID selectors. However, to truly replace the use of the getComputedStyle API, I will consider this case.

**Comment by zhaolixin7** [comment]
@ybzky https://github.com/dream-num/univer-pro/issues/1343

**Review by ybzky — APPROVED** [review]

**Comment by waterhydroxyl** [comment]
I see that there are conflicts in the code. Due to the significant impact of pasting on the experience, can I close this PR, review the relevant changes, and rerun the process? @ybzky

**Comment by ybzky** [comment]
> I see that there are conflicts in the code. Due to the significant impact of pasting on the experience, can I close this PR, review the relevant changes, and rerun the process? @ybzky

ok 😁

**Pull request closed without merging**